### PR TITLE
7.x.metro

### PR DESCRIPTION
**Pull request by willtp87**

### CODE REVIEW

File: islandora_usage_stats.install [hunk 1]
```diff
@@ -2,181 +2,308 @@
...
-                'dsid'
-            )
-        )
-    );
-    return $schema;
+
+/**
+ * Implements hook_schema().
+ *
+ * @todo Link to user table by UID don't make a foreign key.
```
Comment (adam-vessey) [inline islandora_usage_stats.install:11]: Not sure about this... Should probably be a foreign key, and have something to remap to "anonymous" when users are deleted.

File: islandora_usage_stats.module [hunk 2]
```diff
@@ -227,110 +213,95 @@ function islandora_usage_stats_block_contents($delta)
...
+      'description' => t('Manage usage tracking for the Islandora Usage Stats module'),
+    ),
+    'view usage stats block' => array(
+      'title' => t('View usage tracking stats block'),
+      'description' => t('View usage tracking stats block which includes number of views and downloads'),
+    ),
+    ISLANDORA_USAGE_STATS_VIEW_REPORTS => array(
+      'title' => t('View Islandora Usage Reports'),
+      'description' => t('View reports generated by Islandora Usage Statistics,
+        this does not respect XACML.'),
```
Comment (adam-vessey) [inline islandora_usage_stats.module]: Never sure about these... Newlines in strings might get... weird? Dunno.

File: islandora_usage_stats.install (hunk 1 above)
Comment (willtp87) [inline islandora_usage_stats.install:11]: I was thinking there is useful information that would be lost if we did that.  I don't think that Drupal re-uses uids so I though this a good compromise to moving to a more complex system that kept track of specific deleted users.

File: includes/islandora_usage_stats.class.inc [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+    $object_access_id = db_insert('islandora_usage_stats_object_access_log')->fields(array(
+        'pid' => $this->pid,
+        'time' => REQUEST_TIME,
+        'ip' => ip_address(),
+        'user' => ($user->uid) ? $user->name : variable_get('anonymous', t('Anonymous')),
+    ))->execute();
+
+    // Log content models.
+    $models = $object->models;
+    unset($models[array_search('fedora-system:FedoraObject-3.0', $models)]);
```
Comment (adam-vessey) [inline includes/islandora_usage_stats.class.inc]: Seems somewhat unsafe... If we happen to not find `fedora-system:FedoraObject-3.0`, we'd still unset the first value ('cause `FALSE` == 0).

**Pull request merged**